lib/odbc_adapter/adapters/null_odbc_adapter.rb

```diff
@@ -4,15 +4,36 @@ module Adapters
     # registry. This allows for minimal support for DBMSs for which we don't
     # have an explicit adapter.
     class NullODBCAdapter < ActiveRecord::ConnectionAdapters::ODBCAdapter
-      class BindSubstitution < Arel::Visitors::ToSql
-        include Arel::Visitors::BindVisitor
-      end
-
-      # Using a BindVisitor so that the SQL string gets substituted before it is
-      # sent to the DBMS (to attempt to get as much coverage as possible for
+      # Using the generic ToSql visitor (to attempt to get as much coverage as possible for
       # DBMSs we don't support).
       def arel_visitor
-        BindSubstitution.new(self)
+        Arel::Visitors::ToSql.new(self)
+      end
+
+      def self.quote_column_name(name)
+        name = name.to_s
+        quote_char = identifier_quote_char.to_s.strip
+
+        return name if quote_char.length.zero?
+        quote_char = quote_char[0]
+
+        # Avoid quoting any already quoted name
+        return name if name[0] == quote_char && name[-1] == quote_char
+
+        # If upcase identifiers, only quote mixed case names.
+        if upcase_identifiers?
```

Comment:
If this is always true, why do we need this? I recognize that this is the piece of the logic that previously relied on instance state of the connection.

Reply:
We don't need this if it's always true, I am hoping to test this with Snowflake and confirm that this is working as expected and then I can definitely clean this up a bit more

I'm thinking it also makes sense to add some tests around using the null adapter to query postgres (assuming that works) as a sanity check that this adapter isn't totally broken

Reply:
> I'm thinking it also makes sense to add some tests around using the null adapter to query postgres (assuming that works) as a sanity check that this adapter isn't totally broken

Sounds good.

Reply:
We now have the tests (well done, and thank you very much). Have we checked against snowflake? Are we going to remove this conditional? Or retain it and modify the implementation of `upcase_identifiers?`?

Reply:
Oof, thanks for the reminder. I did try this against Snowflake and queried without any issues

Double checking the snowflake docs, I think this is the right configuration so I'll clean up this code a bit